Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -22,8 +23,47 @@ private static void Test(string args, int expected)
 
         public static int FindDigit(string equation)
         {
-            // Add your code here.
-            throw new NotImplementedException();
+            
+            string[] vari = Regex.Split(equation, pattern: "[*=]");
+            
+            var culp = 0;
+            for (int i = 0; i < 3; i++)
+            {
+                if (vari[i].Contains('?'))
+                {
+                    culp = i;
+                    break;
+                }
+            }
+            string res;
+            int A, B, C;
+            if (culp == 0)
+            {
+                B = Int32.Parse(vari[1]);
+                C = Int32.Parse(vari[2]);
+                A = (C / B);
+                res = A.ToString();
+            }
+            else if (culp == 1)
+            {
+                A = Int32.Parse(vari[0]);
+                C = Int32.Parse(vari[2]);
+                B = (C / A);
+                res = B.ToString();
+            }
+            else
+            {
+                A = Int32.Parse(vari[0]);
+                B = Int32.Parse(vari[1]);
+                C = A * B;
+                res = C.ToString();
+            }
+            if ((res.Length != vari[culp].Length) || A*B!=C)

```

Comment:
Could you please add comments to explain what this line of code is doing or maybe you can put this logic in some method whose name will help me to understand the logic behind it.
I understood the part A*B!=C but I didn't understand the 1st condition ie (res.Length != vari[culp].Length)